Makefile

```diff
@@ -161,6 +161,15 @@ e2e-tests-compile:
 # Compile e2e tests for each platform.
 build.code.platform: e2e-tests-compile
 
+# NOTE(hasheddan): the build submodule currently overrides XDG_CACHE_HOME in
+# order to force the Helm 3 to use the .work/helm directory. This causes Go on
+# Linux machines to use that directory as the build cache as well. We should
+# adjust this behavior in the build submodule because it is also causing Linux
+# users to duplicate their build cache, but for now we just make it easier to
+# identify its location in CI so that we cache between builds.
+go.cachedir:
+	@go env GOCACHE
```

Comment:
Dumb make question - how is this different from just calling `go env GOCACHE` directly like we did before?

Reply:
@negz this calls it in our `make` context, which means we have already evaluated the helm 3 override and export of `XDG_CACHE_HOME`. This also gives us the nice property of not breaking things when we fix this upstream in the `build` submodule (it will still just evaluate to whatever it then equals in the new `make` context). And yep it broke with helm 3 :)